source/qml/Samples/QmlVisualizer/AdaptiveCardQml/FilteredChoiceSetRender.qml

```diff
@@ -130,26 +170,130 @@ ComboBox {
         }	
     }
     
-    contentItem: TextInput {
-        id: textField
-        text: ""
-        color:  inputFieldConstants.textColor
-        leftPadding: inputFieldConstants.textHorizontalPadding
-        rightPadding: inputFieldConstants.textHorizontalPadding
-        topPadding: inputFieldConstants.textVerticalPadding + 3.5
-        bottomPadding: inputFieldConstants.textVerticalPadding
+    contentItem: ScrollView {
+        id: textFieldScroller
+
+        property alias buttonFlowModel: comboBox.selectedChoices
+
+        function clearMultiselectTile(index) {
+            comboBox.clearMultiselectTile(index);
+        }
+        clip: true
+        ScrollBar.vertical.policy: ScrollBar.AsNeeded
+        ScrollBar.horizontal.policy: ScrollBar.AlwaysOff
+        contentWidth: width
+        contentHeight: textField.height
+
+        MouseArea {
+            id: mouseArea
         
-        onFocusChanged: {
-            // Open the dropdown when the TextField gets focus
-            if (focus) {
+            anchors.fill: parent
+            onClicked: (mouse)=> {
+                textField.forceActiveFocus();
+            }
+        }                
+
+        TextInput {
+            id: textField
+
+            text: ""
+            clip: true
+            color:  inputFieldConstants.textColor
+            font.family: "Segeo UI"
+            font.pixelSize:  inputFieldConstants.pixelSize
+            leftPadding: textFieldScroller.buttonFlowModel.length == 0 ?  8 : repeater.itemAt(repeater.count - 1).x + repeater.itemAt(repeater.count - 1).width + 16
+            rightPadding: inputFieldConstants.textHorizontalPadding
+            topPadding: textFieldScroller.buttonFlowModel.length == 0 ? 6 : repeater.itemAt(repeater.count - 1).y + 6
+            bottomPadding: inputFieldConstants.textVerticalPadding
+            width: parent.width
+        
+            onFocusChanged: {
+                // Open the dropdown when the TextField gets focus
+                if (focus) {
+                    comboBox.popup.open();
+                }
+            }
+            onTextChanged: {
+                // Open the dropdown when the user types anything
                 comboBox.popup.open();
+                comboBox.filterOptions();
+                textField.forceActiveFocus();
+            }
+            
+            Flow {
+                id: buttonFlow
+                anchors.fill: parent
+                anchors.topMargin: 4
+                anchors.leftMargin: 8
+                spacing: 10 // Adjust the spacing between buttons
+                        
+                Repeater {
```

Comment:
Just wondering, would it make sense to have this in a separate Qml file as component @snbehar ?